### PR TITLE
Clean up stack size definition.

### LINKED ISSUE #23: Need clean up stack size definition
Currently, the stack size definition is at emu - https://github.com/intel/rust-spdm/blob/main/test/spdm-emu/src/lib.rs#L65

```
const TRANSPORT_SIZE: usize = MAX(
    size_of::<PciDoeTransportEncap>(),
    size_of::<MctpTransportEncap>(),
);
const DEVICE_IO_SIZE: usize = size_of::<SocketIoTransport>();
const SPDMLIB_SIZE: usize = size_of::<spdmlib::common::SpdmContext>() +
                            size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1) + size_of::<SpdmCertChainBuffer>() * SPDM_MAX_SLOT_NUMBER + // SpdmProvisionInfo
                            size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1) + // SpdmPeerInfo
                            (MAX_SPDM_MSG_SIZE + SENDER_BUFFER_SIZE + RECEIVER_BUFFER_SIZE) * 5 + // send/receive + encode/decode
                            MAX_SPDM_CERT_CHAIN_DATA_SIZE * 8 + // worst case: 8 slots
                            (MAX_SPDM_MEASUREMENT_RECORD_SIZE + MAX_SPDM_MEASUREMENT_VALUE_LEN) * 255 + // worst case: 255 index
                            MAX_SPDM_PSK_CONTEXT_SIZE + // for PSK
                            MAX_SPDM_PSK_HINT_SIZE + // for PSK
                            size_of::<usize>() * 256; // for general stack case

const EMU_FUNCTION_STACK: usize = SENDER_BUFFER_SIZE
    + RECEIVER_BUFFER_SIZE
    + size_of::<TcpStream>()
    + size_of::<PciDoeTransportEncap>()
    + size_of::<MctpTransportEncap>()
    + size_of::<SpdmMeasurementRecordStructure>() * 255
    + size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1)
    + size_of::<SpdmCertChainBuffer>() * SPDM_MAX_SLOT_NUMBER
    + size_of::<usize>() * 256; // for general stack case

pub const EMU_STACK_SIZE: usize =
    TRANSPORT_SIZE + DEVICE_IO_SIZE + SPDMLIB_SIZE + EMU_FUNCTION_STACK;
```

We should:
1) move SPDMLIB_SIZE to https://github.com/intel/rust-spdm/tree/main/spdmlib.
2) split TRANSPORT_SIZE to MCTP_TRANSPORT_SIZE and PCI_DOE_TRANSPORT_SIZE, then move to https://github.com/intel/rust-spdm/tree/main/mctp_transport and https://github.com/intel/rust-spdm/tree/main/pcidoe_transport.

### PR DESCRIPTION
**Pull request by longlongyang**
fix #23

### CODE REVIEW

**Comment by jyao1** [comment]
Should we use XXX_STACK_SIZE ?

File: test/spdm-emu/src/lib.rs [hunk 1]
```diff
@@ -37,20 +34,8 @@ pub const fn MAX(a: usize, b: usize) -> usize {
...
-                            size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1) + size_of::<SpdmCertChainBuffer>() * SPDM_MAX_SLOT_NUMBER + // SpdmProvisionInfo
-                            size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1) + // SpdmPeerInfo
-                            (MAX_SPDM_MSG_SIZE + SENDER_BUFFER_SIZE + RECEIVER_BUFFER_SIZE) * 5 + // send/receive + encode/decode
-                            MAX_SPDM_CERT_CHAIN_DATA_SIZE * 8 + // worst case: 8 slots
-                            (MAX_SPDM_MEASUREMENT_RECORD_SIZE + MAX_SPDM_MEASUREMENT_VALUE_LEN) * 255 + // worst case: 255 index
-                            MAX_SPDM_PSK_CONTEXT_SIZE + // for PSK
-                            MAX_SPDM_PSK_HINT_SIZE + // for PSK
-                            size_of::<usize>() * 256; // for general stack case
 
 const EMU_FUNCTION_STACK: usize = SENDER_BUFFER_SIZE
```
Comment (jyao1) [inline test/spdm-emu/src/lib.rs]: please use EMU_FUNCTION_STACK_SIZE.

File: spdmlib/src/lib.rs [hunk 2]
```diff
@@ -28,3 +28,14 @@ pub mod secret;
 pub mod time;
 
 pub mod config;
+
+use core::mem::size_of;
+pub const SPDMLIB_SIZE: usize = size_of::<crate::common::SpdmContext>() +
```
Comment (jyao1) [inline spdmlib/src/lib.rs]: SPDM_STACK_SIZE

File: mctp_transport/src/lib.rs [hunk 3]
```diff
@@ -9,3 +9,5 @@ mod header;
 pub use header::*;
 
 extern crate codec;
+
+pub const MCTP_TRANSPORT_SIZE: usize = core::mem::size_of::<MctpTransportEncap>();
```
Comment (jyao1) [inline mctp_transport/src/lib.rs]: MCTP_TRANSPORT_STACK_SIZE

File: pcidoe_transport/src/lib.rs [hunk 4]
```diff
@@ -9,3 +9,5 @@ mod header;
 pub use header::*;
 
 extern crate codec;
+
+pub const PCIDOE_TRANSPORT_SIZE: usize = core::mem::size_of::<PciDoeTransportEncap>();
```
Comment (jyao1) [inline pcidoe_transport/src/lib.rs]: PCIDOE_TRANSPORT_STACK_SIZE

File: test/spdm-emu/src/lib.rs [hunk 5]
```diff
@@ -37,20 +34,8 @@ pub const fn MAX(a: usize, b: usize) -> usize {
     }
 }
 
-const TRANSPORT_SIZE: usize = MAX(
-    size_of::<PciDoeTransportEncap>(),
-    size_of::<MctpTransportEncap>(),
-);
+const TRANSPORT_SIZE: usize = MAX(PCIDOE_TRANSPORT_SIZE, MCTP_TRANSPORT_SIZE);
```
Comment (jyao1) [inline test/spdm-emu/src/lib.rs]: TRANSPORT_STACK_SIZE

**Comment by longlongyang** [comment]
> Should we use XXX_STACK_SIZE ?

good advice. updated.

File: test/spdm-emu/src/lib.rs (hunk 1 above)
Comment (longlongyang) [reply test/spdm-emu/src/lib.rs]: > please use EMU_FUNCTION_STACK_SIZE.

updated.

File: spdmlib/src/lib.rs (hunk 2 above)
Comment (longlongyang) [reply spdmlib/src/lib.rs]: > SPDM_STACK_SIZE

updated.

File: mctp_transport/src/lib.rs (hunk 3 above)
Comment (longlongyang) [reply mctp_transport/src/lib.rs]: > MCTP_TRANSPORT_STACK_SIZE

updated.

File: pcidoe_transport/src/lib.rs (hunk 4 above)
Comment (longlongyang) [reply pcidoe_transport/src/lib.rs]: > PCIDOE_TRANSPORT_STACK_SIZE

updated.

File: test/spdm-emu/src/lib.rs (hunk 5 above)
Comment (longlongyang) [reply test/spdm-emu/src/lib.rs]: > TRANSPORT_STACK_SIZE

updated.

**Pull request merged**